### PR TITLE
Adding Recovery Exit Support for Composable Stable and Weighted Pools;

### PR DESCRIPTION
**Pull request by lgahdl**
- Composable Stable;
- Weighted;

### CODE REVIEW

File: test/lib/utils/addresses.ts [hunk 1]
```diff
@@ -25,6 +25,27 @@ export const TOKENS: Record<number, Record<string, TestToken>> = {
             decimals: 6,
             slot: 2,
         },
+        swETH_bb_a_WETH_BPT: {
+            address: '0x156c02f3f7fef64a3a9d80ccf7085f23cce91d76',
+            decimals: 18,
+            slot: 0,
+        },
```
Comment (brunoguerios) [inline test/lib/utils/addresses.ts:32]: Given this has the same content as the pool, we could avoid duplicates by using the pool only - what do you think?

File: test/v2/removeLiquidity/composableStable.recovery.integration.test.ts [hunk 2]
```diff
@@ -105,29 +105,22 @@ export class MockApi {
     public async getPool(id: Hex): Promise<PoolState> {
         const tokens = [
             {
-                address:
-                    '0x156c02f3f7fef64a3a9d80ccf7085f23cce91d76' as Address, // swETH-bb-a-WETH-BPT
-                decimals: 18,
+                ...TOKENS_MAINNET.swETH_bb_a_WETH_BPT,
```
Comment (brunoguerios) [inline test/v2/removeLiquidity/composableStable.recovery.integration.test.ts:108]: I considered using the spread operator here, but I suggest this small change so we're more explicit from what we expect from the API.
```suggestion
                address: TOKENS_MAINNET.swETH_bb_a_WETH_BPT.address,
                decimals: TOKENS_MAINNET.swETH_bb_a_WETH_BPT.decimals,
```

File: test/v2/removeLiquidity/composableStable.recovery.integration.test.ts [hunk 3]
```diff
@@ -105,29 +105,22 @@ export class MockApi {
...
-                address:
-                    '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2' as Address, // swETH
-                decimals: 18,
+                ...TOKENS_MAINNET.swETH,
                 index: 2,
             },
         ];
 
         return {
+            ...POOLS[chainId].swETH_bb_a_WETH_BPT,
```
Comment (brunoguerios) [inline test/v2/removeLiquidity/composableStable.recovery.integration.test.ts:122]: Same here with the spread operator.

File: test/v2/removeLiquidity/composableStable.recovery.integration.test.ts [hunk 4]
```diff
@@ -20,17 +19,18 @@ import {
...
 } from 'test/lib/utils/removeLiquidityHelper';
 import { test } from 'vitest';
+import { POOLS, TOKENS } from 'test/lib/utils/addresses';
 
 const chainId = ChainId.MAINNET;
 const { rpcUrl } = await startFork(ANVIL_NETWORKS.MAINNET);
-const poolId =
-    '0x156c02f3f7fef64a3a9d80ccf7085f23cce91d76000000000000000000000570'; // swETH-bb-a-WETH-BPT
+const TOKENS_MAINNET = TOKENS[chainId];
+const POOLS_MAINNET = POOLS[chainId];
```
Comment (brunoguerios) [inline test/v2/removeLiquidity/composableStable.recovery.integration.test.ts:33]: Something that could make it cleaner and easier to replace the pool if needed would be to declare a testPool and reuse through the test:
```suggestion
const testPool = POOLS[chainId].swETH_bb_a_WETH_BPT;
```

File: test/lib/utils/addresses.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { Address, Hex, PoolType } from '@/types';
 import { ChainId } from '@/utils';
 
-export type TestToken = { address: Address; decimals: number; slot: number };
+export type TestToken = { address: Address; decimals: number; slot?: number };
```
Comment (brunoguerios) [inline test/lib/utils/addresses.ts:4]: This change will affect other tests. I thought the linter would pick it up, but apparently it didn't.
Can you please check the addLiquidityNested tests that use these new constants?

Comment (lgahdl) [reply test/lib/utils/addresses.ts:4]: Yeah there was an error indeed, I fixed the type as number[] for the addLiquidityNested tests;

**Review by brunoguerios — APPROVED** [review]
LGTM 👍

**Comment by brunoguerios** [comment]
Feel free to add changeset and merge ✅

**Pull request merged**